src/internal/starlark/reflect.go

```diff
@@ -0,0 +1,240 @@
+package starlark
+
+import (
+	"fmt"
+	"math/big"
+	"reflect"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/zeebo/xxh3"
+	"go.starlark.net/starlark"
+)
+
+// Reflect is a starlark.Value that exposes the fields of struct types.  For example, `package foo;
+// struct Test { Foo string }` would be of type "foo.Test" and have one attribute, "Foo", in
+// starlark, used like value.Foo.
+type Reflect struct {
+	Any any
+}
+
+var _ starlark.Value = (*Reflect)(nil)
+var _ starlark.Mapping = (*Reflect)(nil)
+var _ starlark.HasAttrs = (*Reflect)(nil)
+var _ starlark.TotallyOrdered = (*Reflect)(nil)
+
+func (r *Reflect) String() string {
+	if r.Any == nil {
+		return "<nil reflect>"
+	}
+	if s, ok := r.Any.(fmt.Stringer); ok {
+		return s.String()
+	}
+	return fmt.Sprintf("%#v", r.Any)
+}
+func (r *Reflect) Type() string {
+	return reflect.TypeOf(r.Any).String()
+}
+func (r *Reflect) Freeze() {}
+func (r *Reflect) Hash() (uint32, error) {
+	if r.Any == nil {
+		return 0, errors.New("nil object in Reflect{}")
+	}
+	return uint32(xxh3.HashString(r.String())), nil
+}
+func (r *Reflect) Truth() starlark.Bool { return r.Any == nil }
```

Comment:
Idk how Truth is used, but should it be `r.Any != nil`?

Reply:
Yup.